packages/ExpandCollapse/PanelWrapper/PanelWrapper.modules.scss

```diff
@@ -16,5 +17,21 @@
 }
 
 .alignSelfFlexEnd {
-  align-self: flex-end;
+  @include mq($until: md) {
+    align-self: flex-end;
+    span:nth-child(1) {
+      display: none;
+    }
+    span:nth-child(2) {
+      display: initial;
+    }
+  }
+  @include mq($from: md) {
+    span:nth-child(1) {
+      display: initial;
+    }
+    span:nth-child(2) {
+      display: none;
+    }
+  }
```

Comment:
We don't typically target element selectors in Sass as a convention and best practice (`span:nth-child(1)`), and don't encourage our consumers to do so either. This kind of styles makes it very hard to tell what is going on without having full knowledge of both the markup and the styles.

What do you think about trying it this way instead?

```jsx
<span className={styles.alignSelfFlexEnd}>
  <span className={styles.showFromMd}>
    <Text data-testid="tertiarytext" size="large">{tertiaryText}</Text>
  </span>
  <span className={styles.showUntilMd}>
    <Text data-testid="tertiarytext" size="large">{tertiaryText}</Text>
  </span>
</span>
```

```scss
.showFromMd {
  display: none;

  @include mq($from: md) {
    display: initial;
  }
}

.showUntilMd {
  display: none;

  @include mq($until: md) {
    display: initial;
  }
}
```

Reply:
I considered this approach at first but ended up deciding against it because of the extra spans and classes being added in. (I wanted to add as little as possible since this is only a temporary fix) That being said, if you guys tend to not use `nth-child` then I'll switch over to this approach.

Reply:
Thanks for adding your context.

> because of the extra spans and classes being added in

We have not really been worried about the presence of the "extra" spans, as we would want to measure the impact of that before trying to optimize it. I think the wrapping spans in this case are worth it.

Re: the `nth-child` approach.

One of the big concerns here is that it is completely dependent on the _order_ of the children elements. If someone comes by and rearranges the order of those 2 `<Text>` components for whatever reason, the behaviour would be broken and we wouldn't have any real way of detecting that. The wrapping spans with classes is both more explicit, and less brittle.

Reply:
Yeah, that's fair enough to want things to be more explicit. I'll be sure to keep that in mind for next time. 😄